pulp_container/app/registry_api.py

```diff
@@ -392,10 +392,19 @@ def get_dr_push(self, request, path, create=False):
 
     def create_dr(self, path, request):
         with transaction.atomic():
-            repository = serializers.ContainerPushRepositorySerializer.get_or_create({"name": path})
-            distribution = serializers.ContainerDistributionSerializer.get_or_create(
-                {"base_path": path, "name": path}, {"repository": get_url(repository)}
-            )
+            try:
+                repository = serializers.ContainerPushRepositorySerializer.get_or_create(
+                    {"name": path}
+                )
+            except models.ContainerPushRepository.DoesNotExist:
+                raise RepositoryInvalid(name=path, message="Repository is read-only.")
+
+            try:
+                distribution = serializers.ContainerDistributionSerializer.get_or_create(
+                    {"base_path": path, "name": path}, {"repository": get_url(repository)}
+                )
+            except models.ContainerDistribution.DoesNotExist:
+                raise RepositoryInvalid(name=path, message="Repository is ready-only.")
```

Comment:
```suggestion
            try:
                repository = serializers.ContainerPushRepositorySerializer.get_or_create(
                    {"name": path}
                )
                distribution = serializers.ContainerDistributionSerializer.get_or_create(
                    {"base_path": path, "name": path}, {"repository": get_url(repository)}
                )
            except ObjectDoesNotExist:
                raise RepositoryInvalid(name=path, message="Repository is ready-only.")
```

Reply:
:man_facepalming:

Reply:
hum... actually, it seems like it does not work:
```
...
   File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 637, in get
     raise self.model.DoesNotExist(
 pulp_container.app.models.ContainerDistribution.DoesNotExist: ContainerDistribution matching query does not exist.
...
   File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 637, in get
     raise self.model.DoesNotExist(
 pulp_container.app.models.ContainerPullThroughRemote.DoesNotExist: ContainerPullThroughRemote matching query does not exist.
```

the django `ObjectDoesNotExist` exception did not catch the `ContainerDistribution.DoesNotExist` and/or `ContainerPullThroughRemote.DoesNotExist` exceptions

Reply:
nvm, the above exception is not related with the proposed change.

We opened a new issue (#1729) to investigate it.